### ISSUE TITLE
[Bug]: Shutdown problem when we use ADAG

### ISSUE THREAD

**Issue by Bye-legumes** [vllm-project/vllm#8208]
### Your current environment

```
PyTorch version: 2.4.0+cu121
Is debug build: False
CUDA used to build PyTorch: 12.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.2 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: version 3.30.2
Libc version: glibc-2.35

Python version: 3.11.9 (main, Apr 19 2024, 16:48:06) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-6.2.0-37-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: 12.2.128
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration:
GPU 0: NVIDIA RTX A6000
GPU 1: NVIDIA RTX A6000
GPU 2: NVIDIA RTX A6000
GPU 3: NVIDIA RTX A6000

Nvidia driver version: 530.30.02
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Address sizes:                      48 bits physical, 48 bits virtual
Byte Order:                         Little Endian
CPU(s):                             64
On-line CPU(s) list:                0-63
Vendor ID:                          AuthenticAMD
Model name:                         AMD Ryzen Threadripper PRO 5975WX 32-Cores
CPU family:                         25
Model:                              8
Thread(s) per core:                 2
Core(s) per socket:                 32
Socket(s):                          1
Stepping:                           2
Frequency boost:                    enabled
CPU max MHz:                        7006.6401
CPU min MHz:                        1800.0000
BogoMIPS:                           7187.11
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 invpcid_single hw_pstate ssbd mba ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd amd_ppin brs arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif v_spec_ctrl umip pku ospke vaes vpclmulqdq rdpid overflow_recov succor smca fsrm
Virtualization:                     AMD-V
L1d cache:                          1 MiB (32 instances)
L1i cache:                          1 MiB (32 instances)
L2 cache:                           16 MiB (32 instances)
L3 cache:                           128 MiB (4 instances)
NUMA node(s):                       1
NUMA node0 CPU(s):                  0-63
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Not affected
Vulnerability Retbleed:             Not affected
Vulnerability Spec rstack overflow: Mitigation; safe RET, no microcode
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Retpolines, IBPB conditional, IBRS_FW, STIBP always-on, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] nvidia-cublas-cu12==12.1.3.1
[pip3] nvidia-cuda-cupti-cu12==12.1.105
[pip3] nvidia-cuda-nvrtc-cu12==12.1.105
[pip3] nvidia-cuda-runtime-cu12==12.1.105
[pip3] nvidia-cudnn-cu12==9.1.0.70
[pip3] nvidia-cufft-cu12==11.0.2.54
[pip3] nvidia-curand-cu12==10.3.2.106
[pip3] nvidia-cusolver-cu12==11.4.5.107
[pip3] nvidia-cusparse-cu12==12.1.0.106
[pip3] nvidia-ml-py==12.560.30
[pip3] nvidia-nccl-cu12==2.20.5
[pip3] nvidia-nvjitlink-cu12==12.6.20
[pip3] nvidia-nvtx-cu12==12.1.105
[pip3] pyzmq==26.2.0
[pip3] torch==2.4.0
[pip3] torchvision==0.19.0
[pip3] transformers==4.44.1
[pip3] triton==3.0.0
[conda] numpy                     1.26.4                   pypi_0    pypi
[conda] nvidia-cublas-cu12        12.1.3.1                 pypi_0    pypi
[conda] nvidia-cuda-cupti-cu12    12.1.105                 pypi_0    pypi
[conda] nvidia-cuda-nvrtc-cu12    12.1.105                 pypi_0    pypi
[conda] nvidia-cuda-runtime-cu12  12.1.105                 pypi_0    pypi
[conda] nvidia-cudnn-cu12         9.1.0.70                 pypi_0    pypi
[conda] nvidia-cufft-cu12         11.0.2.54                pypi_0    pypi
[conda] nvidia-curand-cu12        10.3.2.106               pypi_0    pypi
[conda] nvidia-cusolver-cu12      11.4.5.107               pypi_0    pypi
[conda] nvidia-cusparse-cu12      12.1.0.106               pypi_0    pypi
[conda] nvidia-ml-py              12.560.30                pypi_0    pypi
[conda] nvidia-nccl-cu12          2.20.5                   pypi_0    pypi
[conda] nvidia-nvjitlink-cu12     12.6.20                  pypi_0    pypi
[conda] nvidia-nvtx-cu12          12.1.105                 pypi_0    pypi
[conda] pyzmq                     26.2.0                   pypi_0    pypi
[conda] torch                     2.4.0                    pypi_0    pypi
[conda] torchvision               0.19.0                   pypi_0    pypi
[conda] transformers              4.44.1                   pypi_0    pypi
[conda] triton                    3.0.0                    pypi_0    pypi
ROCM Version: Could not collect
Neuron SDK Version: N/A
vLLM Version: 0.5.5@09c7792610ada9f88bbf87d32b472dd44bf23cc2
vLLM Build Flags:
CUDA Archs: Not Set; ROCm: Disabled; Neuron: Disabled
GPU Topology:
GPU0    GPU1    GPU2    GPU3    CPU Affinity    NUMA Affinity
GPU0     X      SYS     SYS     NV4     0-63            N/A
GPU1    SYS      X      NV4     SYS     0-63            N/A
GPU2    SYS     NV4      X      SYS     0-63            N/A
GPU3    NV4     SYS     SYS      X      0-63            N/A

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

```

### 🐛 Describe the bug

When executor shutdown it will has the problem.
The reason is the interpreter is during shutdown so can not import module and the module imported previously is None
```
(RayWorkerWrapper pid=1706125) INFO 09-03 12:01:38 custom_all_reduce.py:223] Registering 2275 cuda graph addresses
(RayWorkerWrapper pid=1706125) INFO 09-03 12:01:38 model_runner.py:1300] Graph capturing finished in 13 secs.
Exception ignored in: <function LLMEngine.__del__ at 0x7f2ff66af060>
Traceback (most recent call last):
  File "/home/zhilong/miniconda3/envs/vllm/lib/python3.11/site-packages/vllm/engine/llm_engine.py", line 492, in __del__
  File "/home/zhilong/miniconda3/envs/vllm/lib/python3.11/site-packages/vllm/executor/ray_gpu_executor.py", line 72, in shutdown
  File "/home/zhilong/miniconda3/envs/vllm/lib/python3.11/site-packages/ray/dag/compiled_dag_node.py", line 1707, in teardown
  File "/home/zhilong/miniconda3/envs/vllm/lib/python3.11/site-packages/ray/dag/compiled_dag_node.py", line 1505, in teardown
  File "/home/zhilong/miniconda3/envs/vllm/lib/python3.11/logging/__init__.py", line 1489, in info
  File "/home/zhilong/miniconda3/envs/vllm/lib/python3.11/logging/__init__.py", line 1634, in _log
  File "/home/zhilong/miniconda3/envs/vllm/lib/python3.11/logging/__init__.py", line 1644, in handle
  File "/home/zhilong/miniconda3/envs/vllm/lib/python3.11/logging/__init__.py", line 1706, in callHandlers
  File "/home/zhilong/miniconda3/envs/vllm/lib/python3.11/logging/__init__.py", line 978, in handle
  File "/home/zhilong/miniconda3/envs/vllm/lib/python3.11/site-packages/ray/_private/log.py", line 52, in emit
ImportError: sys.meta_path is None, Python is likely shutting down
Exception ignored in: <function RayGPUExecutor.__del__ at 0x7f30b3ac05e0>
Traceback (most recent call last):
  File "/home/zhilong/miniconda3/envs/vllm/lib/python3.11/site-packages/vllm/executor/ray_gpu_executor.py", line 484, in __del__
  File "/home/zhilong/miniconda3/envs/vllm/lib/python3.11/site-packages/vllm/executor/ray_gpu_executor.py", line 72, in shutdown
  File "/home/zhilong/miniconda3/envs/vllm/lib/python3.11/site-packages/ray/dag/compiled_dag_node.py", line 1707, in teardown
  File "/home/zhilong/miniconda3/envs/vllm/lib/python3.11/site-packages/ray/dag/compiled_dag_node.py", line 1502, in teardown
  File "/home/zhilong/miniconda3/envs/vllm/lib/python3.11/site-packages/ray/dag/compiled_dag_node.py", line 1472, in wait_teardown
AttributeError: 'NoneType' object has no attribute 'GetTimeoutError'"
```

Reproduce script
export DISTRIBUTED_EXECUTOR_BACKEND=ray
export VLLM_USE_RAY_SPMD_WORKER=1
export VLLM_USE_RAY_COMPILED_DAG=1
```
import time
from vllm import LLM, SamplingParams

# Set the tensor parallelism size
tensor_parallelism_size = 2

# Initialize the model
model_path = "/home/models/Llama-2-7b-hf"
llm = LLM(model=model_path, tensor_parallel_size=tensor_parallelism_size,distributed_executor_backend="ray")

# Define the batch of prompts
prompts = [
    "Once upon a time, in a land far, far away,",
    "The quick brown fox jumps over the lazy dog.",
    "In the midst of the dense forest, a lonely wolf howled at the moon.",
    "To be, or not to be, that is the question.",
    "Artificial intelligence is transforming the world."
]

# Set sampling parameters
sampling_params = SamplingParams(max_tokens=50, temperature=0.7)

# Warm-up the model with a simple inference
warmup_prompt = ["Warm-up prompt"]
llm.generate(warmup_prompt, sampling_params)

# Measure inference time
start_time = time.time()

# Perform batch inference
outputs = llm.generate(prompts*100, sampling_params)

# Calculate the number of tokens generated
total_tokens = sum(len(completion.token_ids) for output in outputs for completion in output.outputs)

# Calculate the total time taken for inference
total_time = time.time() - start_time

# Calculate tokens per second
tokens_per_second = total_tokens / total_time

print(f"Total tokens generated: {total_tokens}")
print(f"Total time taken (seconds): {total_time:.4f}")
print(f"Tokens per second: {tokens_per_second:.2f}")
```

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by rkooo567** [comment]
we will address this issue soon!

**Comment by Bye-legumes** [comment]
Thanks fix! just closed!
